packages/deployments/contracts/contracts/core/connext/facets/SwapAdminFacet.sol

```diff
@@ -176,6 +185,35 @@ contract SwapAdminFacet is BaseConnextFacet {
     emit SwapInitialized(_key, entry, msg.sender);
   }
 
+  /**
+   * @notice remove Swap Struct for key
+   *
+   * @param _key the hash of the canonical id and domain for token
+   */
+  function removeSwap(bytes32 _key) external onlyOwnerOrAdmin {
+    uint256 numPooledTokens = s.swapStorages[_key].pooledTokens.length;
+
+    if (numPooledTokens == 0) revert SwapAdminFacet__removeSwap_notInitialized();
+
+    for (uint256 i; i < numPooledTokens; ) {
+      if (s.swapStorages[_key].balances[i] == 0) {
+        revert SwapAdminFacet__removeSwap_NonZeroBalance();
+      }
+
+      if (s.swapStorages[_key].adminFees[i] > 0) {
+        s.swapStorages[_key].withdrawAdminFees(msg.sender);
```

Comment:
`withdrawAdminFees()` also loops over `pooledTokens`, so could be outside of the `for` loop